IPython/html/static/notebook/js/cell.js

```diff
@@ -76,7 +75,12 @@ define([
 
         // load this from metadata later ?
         this.user_highlight = 'auto';
-        this.cm_config = config.cm_config;
+
+        var class_conf_value = {};
+        if(this.class_config){
+            class_conf_value = this.class_config.get_sync('cm_config');
+        }
+        this.cm_config = utils.mergeopt({}, config.cm_config, class_conf_value);
```

Comment:
The naming is getting really confusing here: `class_conf_value` is the value the user has set, if it has arrived, not the value set in the class definition. But if the user config hasn't been loaded yet, both will be the value set on the class, so you'll be merging a dict with itself.

I recommend that we redo this, moving cm_config out of options_default to a separate class-level attribute `default_cm_config`. Then this would look like:

```javascript
utils.mergeopt({}, this.default_cm_config, this.class_config.get_sync('cm_config'))
```